### PR TITLE
Cleanup code package structure and file names

### PR DESCRIPTION
**Pull request by ahg-g**
Cleans up pkg structure and file names. Mostly following Kueue and JobSet.

This PR is a no op, renaming and moving files.

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: test/integration/webhook/leaderworkerset_test.go [hunk 1]
```diff
@@ -12,7 +12,8 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package webhook
```
Comment (kerthcet) [inline test/integration/webhook/leaderworkerset_test.go:15]: nit: still webhooks? We will not import this package. No name conflicts.

File: test/integration/controller/leaderworkerset_test.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package controllertest
```
Comment (kerthcet) [inline test/integration/controller/leaderworkerset_test.go:14]: Nit: package controllers? And rename the folder to controllers for consistency.

File: test/integration/webhook/leaderworkerset_test.go (hunk 1 above)
Comment (kerthcet) [reply test/integration/webhook/leaderworkerset_test.go:15]: Rename the folder to webhooks.

Comment (ahg-g) [reply test/integration/webhook/leaderworkerset_test.go]: done.

File: test/integration/controller/leaderworkerset_test.go (hunk 2 above)
Comment (ahg-g) [reply test/integration/controller/leaderworkerset_test.go]: done.

**Pull request merged**